src/awkward/_connect/jax/reducers.py

```diff
@@ -261,7 +328,7 @@ def apply(
         if array.dtype.kind == "M":
             raise TypeError(f"cannot compute the sum (ak.sum) of {array.dtype!r}")
 
```

Comment:
@ikrommyd - if we want to allow sum of boolean, I think, we should view the data as integers here:

```python
    if array.dtype == np.bool_:
        data = array.data.astype(jax.numpy.int32)
    else:
        data = array.data

```